operator/v1/types_console.go

```diff
@@ -202,6 +205,52 @@ type AddPage struct {
 	DisabledActions []string `json:"disabledActions,omitempty"`
 }
 
+type PerspectiveState string
+
+// These are valid perspective states. "Enabled" means the perspective is shown.
+// "Disabled" means the Perspective is hidden.
+// "AccessReview" means access review check is required to show or hide a Perspective.
+const (
+	Enabled      PerspectiveState = "Enabled"
+	Disabled     PerspectiveState = "Disabled"
+	AccessReview PerspectiveState = "AccessReview"
+)
+
+// `required` and  `missing` can work together esp. in the case where cluster admin
+// wants to show another perspective to users without specific permissions. If both `required` and `missing`
+// are not specified for `state: AccessReview`, access review check will be ignored and perspective will be enabled.
+type PerspectiveAccessReview struct {
+	// required defines a list of permission checks. The perspective will only be shown when all checks are successful. If no checks are specified then we simply ignore the check even if the state is AccessReview.
+	// +optional
+	Required []authorizationv1.ResourceAttributes `json:"required,omitempty"`
+	// missing defines a list of permission checks. The perspective will only be shown when at least one check fails. If no checks are specified then we simply ignore the check even if the state is AccessReview.
+	// +optional
+	Missing []authorizationv1.ResourceAttributes `json:"missing,omitempty"`
+}
+
+// PerspectiveVisibility defines the criteria to show/hide a perspective
+// +union
+type PerspectiveVisibility struct {
+	// state defines the perspective is enabled or disabled or access review check is required.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:=Enabled;Disabled;AccessReview
```

Comment:
To allow discoverability, we may need to include the `""` as a valid value. Can you remove the omitempty from all of the fields in this API except the AccessReview field below, then take the API for a test run and check the validations work?

Reply:
Ignore this comment, doesn't apply as this is an item in a list